include/buffer_tree.h

```diff
@@ -100,16 +105,11 @@ class BufferTree {
   insert_ret_t insert(update_t upd);
 
   /*
-   * Get data from the buffertree given a description of where the data is
-   * @param  task   Where the data we want to extract can be found and which key
-   * @retuns a vector of updates associated with the key
+   * Ask the buffer tree for data and sleep if necessary until it is available.
+   * @param data       this is where to the key and vector of updates associated with it
+   * @return           true true if got valid data, false if unable to get data.
    */
-  data_ret_t get_data(work_t task);
-
-  // queue of work which needs to be done and the locks which control access to it
-  std::queue<work_t> work_queue;
-  std::mutex queue_lock;
-  std::condition_variable queue_cond;
+  bool get_data(data_ret_t &data);
```

Comment:
should this be a pointer instead of a reference?

Reply:
I think it could be either, but going off the conversation we had way back when about toku and std::vector I decided a reference would probably be better